crates/bifrost/src/loglets/local_loglet/log_store_writer.rs

```diff
@@ -161,27 +170,34 @@ impl LogStoreWriter {
         write_batch.put_cf(data_cf, &key.to_bytes(), data);
     }
 
-    fn commit(&mut self, opts: &LocalLogletOptions, write_batch: WriteBatch) {
-        let mut rocksdb_write_options = rocksdb::WriteOptions::new();
-        rocksdb_write_options.disable_wal(opts.rocksdb.rocksdb_disable_wal());
+    async fn commit(&mut self, opts: &LocalLogletOptions, write_batch: WriteBatch) {
+        let mut write_opts = rocksdb::WriteOptions::new();
+        write_opts.disable_wal(opts.rocksdb.rocksdb_disable_wal());
 
         trace!(
             "Committing local loglet current write batch: {} items",
             write_batch.len(),
         );
+        let result = self
+            .rocksdb
+            .write_batch(Priority::High, IoMode::Default, write_opts, write_batch)
+            .await;
 
-        if let Err(e) = self.db.write_opt(&write_batch, &rocksdb_write_options) {
+        if let Err(e) = result {
             error!("Failed to commit local loglet write batch: {}", e);
             self.send_acks(Err(Error::LogStoreError(e.into())));
             return;
         }
 
         if self.manual_wal_flush {
-            if let Err(e) = self.db.flush_wal(true) {
+            // WAL flush is done in the forground, but sync will happen in the background to avoid
+            // blocking IO.
+            if let Err(e) = self.rocksdb.flush_wal(false) {
                 warn!("Failed to flush rocksdb WAL in local loglet : {}", e);
                 self.send_acks(Err(Error::LogStoreError(e.into())));
                 return;
             }
+            self.rocksdb.run_bg_wal_sync();
```

Comment:
Why is it safe to not wait for the sync to have happened before sending the acks back? What would we lose if we awaited the wal sync?

Reply:
We'd lose ~20% of throughout with batching (it'd be much more loss with smaller batches) and we will have to perform all writes on the background thread pool to avoid blocking tokio runtime.

I can make this a configuration option that's enabled by default (as in, sync=true by default) if we want to be stringent by default. What do you think?

Reply:
I've added the option.

Reply:
Sounds like a good compromise to me.